src/Plugin/Validators/ValidDataFile.php

```diff
@@ -195,10 +273,161 @@ public function validateFile(int|null $fid) {
+    $combined_tokens = array_merge($default_tokens, $tokens);
+
+    $container = \Drupal::getContainer();
+
+    // Get the current user in case we trigger a case that needs to log a
+    // message to the administrator.
+    $username = $container->get('current_user')->getAccountName();
+    $service_TripalLogger = $container->get('tripal.logger');
+
+    // Define our items array.
+    $items = [];
+
+    if (($validation_status['case'] == self::$mapping['case-invalid-fid']['dev-case']) ||
+        ($validation_status['case'] == self::$mapping['case-failed-fid']['dev-case'])) {
+
+      $message = $combined_tokens['case-invalid-fid'];
+      // Get the fid of the uploaded file.
+      $fid = $validation_status['failedItems']['fid'];
+
+      // Log a message for the administrator to help with debugging the issue.
+      $service_TripalLogger
+        ->info("The user $username uploaded a file with FID $fid using the Traits Importer, but could not import it as something is wrong with the filename/FID. More specifically, the case message '" . $validation_status['case'] . "' was reported.");
+    }
+    elseif ($validation_status['case'] == self::$mapping['case-empty-file']['dev-case']) {
+      $message = $combined_tokens['case-empty-file'];
+      $items = [
+        'Filename: ' . $validation_status['failedItems']['filename'],
+      ];
+    }
+    elseif (($validation_status['case'] == self::$mapping['case-unsupported-mime']['dev-case']) ||
+            ($validation_status['case'] == self::$mapping['case-unsupported-extension']['dev-case'])) {
+
+      $message = $combined_tokens['case-unsupported-mime'];
+      // Give more info to the user AND log a message to the administrator using
+      // these failed items:
+      $file_mime = $validation_status['failedItems']['mime'];
+      $file_extension = $validation_status['failedItems']['extension'];
+
+      $combined_tokens['file-mime'] = '"' . $file_mime . '"';
+      $combined_tokens['file-extension'] = '"' . $file_extension . '"';
+      $items = [
+        $combined_tokens['explain-unsupported-mime'],
+      ];
+
+      // Log a message for the administrator to help with debugging the issue.
+      $service_TripalLogger
+        ->info("The user $username uploaded a file to the Traits Importer with file extension \"$file_extension\" and mime type \"$file_mime\"");
+    }
+    elseif ($validation_status['case'] == self::$mapping['case-locked-file']['dev-case']) {
+      $message = $combined_tokens['case-locked-file'];
+      $filename = $validation_status['failedItems']['filename'];
+      $fid = $validation_status['failedItems']['fid'];
+      $items = [
+        'Filename: ' . $filename,
+      ];
+
+      // Log a message for the administrator to help with debugging the issue.
+      $service_TripalLogger
+        ->info("The user $username uploaded a file with FID $fid using the Traits Importer, but the file could not be opened using \'@fopen\'. Filename was '$filename'.");
```

Comment:
I would make this an @todo unless we want to address this before merging this PR
```suggestion
    // @todo - determine if we want to include tokens in the items.
```

Reply:
I'm guessing the original line of code disappeared since my initial review comment 🤔

Reply:
The line was removed because it was determined that there would tokens in items with this comment https://github.com/TripalCultivate/TripalCultivate/pull/69#discussion_r2223292711. I apologize for that.

Reply:
Oh that makes sense! This is resolved then 👍